### PR TITLE
Detailed changelog

### LINKED ISSUE #116: Update ChangeLog
Please keep `CHANGELOG.md` up to date with what's changed as new PRs are merged.

### PR DESCRIPTION
**Pull request by TotallyNotChase**
Here's the promised detailed changelog!

Resolves #116

### CODE REVIEW

**Comment by TotallyNotChase** [comment]
I hope I didn't miss anything.....

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,23 +2,190 @@
...
+- Mutually recursive data types, scott encoded records and more provided by `Plutarch.Rec`!
+
+  Module: `Plutarch.Rec`
+
+  Worked on through:
+  * [#60](https://github.com/Plutonomicon/plutarch/pull/60)
+  * [#114](https://github.com/Plutonomicon/plutarch/pull/114)
+  * [#125](https://github.com/Plutonomicon/plutarch/pull/125)
+  * [#175](https://github.com/Plutonomicon/plutarch/pull/175)
+- The Plutus V1 ledger api implemented in Plutarch!
```
Comment (srid) [inline CHANGELOG.md:147]: Should probably say 'api types' - since we don't have functions yet.

**Review by srid — DISMISSED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,23 +2,190 @@
...
-- `do`-notation (!).
-- Benchmarks.
-- `letrec` and records.
-- More complete `Plutarch.Api.V1`.
-- Add generic deriving for `PIsDataRepr`
+- Repo changes (not visible on the user side)
+  - The Plutarch repo has moved to GHC 9.2.1 (GHC 8.10.7 should still be able to depend on Plutarch).
+
+    Relevant PR: [#86](https://github.com/Plutonomicon/plutarch/pull/86)
+  - Major nix updates.
```
Comment (L-as) [inline CHANGELOG.md:9]: These are visible.

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,23 +2,190 @@
...
+  <summary> Breakdown of additions </summary>
+
+  - Boolean utilities:- `pnot`, `#&&`/`pand`/`pand'`, `#||`/`por`/`por'`
+
+    Module: `Plutarch.Bool`
+
+    Added in: [#30](https://github.com/Plutonomicon/plutarch/pull/30)
+  - ByteString utilities:- `pconsBS`, `psliceBS`, `plengthBS`, `pindexBS`
+
+    Module: `Plutarch.ByteString`
```
Comment (L-as) [inline CHANGELOG.md:51]: Why have all of this `Module:` stuff?

**Review by L-as — APPROVED** [review]
LGTM

File: CHANGELOG.md (hunk 1 above)
Comment (TotallyNotChase) [reply CHANGELOG.md]: Agreed, updated!

File: CHANGELOG.md (hunk 2 above)
Comment (TotallyNotChase) [reply CHANGELOG.md:9]: Hmm, true - reworded it a bit.

File: CHANGELOG.md (hunk 3 above)
Comment (TotallyNotChase) [reply CHANGELOG.md:54]: Easy for people to *see* the new changes and the type signatures for now. until we have some hosted haddocks to point to or similar. Also see: https://github.com/Plutonomicon/plutarch/pull/99#discussion_r786905746

**Comment by L-as** [comment]
Maybe you want to update this for #189 and then we can merge this and staging?

**Review by L-as — APPROVED** [review]
LGTM

**Pull request merged**